File: week1/problemset1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{

    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    for (int i = 1; i <= height; i++)
    {
        print_row(height - i, i);
    }
}

void print_row(int spaces, int bricks)
{
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }

    for (int j = 0; j < bricks; j++)
    {
        printf("#");
    }
    printf("\n");
}

File: week6/problemset/sentimental-mario-more/mario.py
def main():

    height = get_height()

    print_pyramide(height)


def print_pyramide(height):
    for i in range(height):
        print_row(height - i - 1, i + 1)


def print_row(spaces, blocks):
    print(" " * spaces + "#" * blocks + "  " + "#" * blocks)


def get_height():
    while True:

        height = get_int("Height: ")

        if height > 0 and height <= 8:
            break

        print("Height must be a positive integer less or equal to 8")

    return height


def get_int(prompt):
    while True:
        try:
            return int(input(prompt))
        except ValueError:
            print("Not an Integer")


main()
